src/server/global_notifier.cpp

```diff
@@ -395,11 +395,11 @@ std::unordered_map<std::string, ObjectChangeSet> const& GlobalNotifier::ChangeNo
     auto realm = Realm::get_shared_realm(config);
 
     Realm::Internal::begin_read(*realm, m_old_version);
-    Group const& g = realm->read_group();
+    Group& g = realm->read_group();
 
     _impl::TransactionChangeInfo info;
     info.track_all = true;
-    _impl::transaction::advance(realm->transaction(), info, m_new_version);
+    _impl::transaction::advance(static_cast<Transaction&>(g), info, m_new_version);
```

Comment:
This cast will do the correct thing, but this requires knowledge of the implementation. Why not operate on `realm->transaction()` here?

Reply:
Regardless of how it gets to the Transaction, the global notifier is operating on implementation details that it's not supposed to know about. Exposing `transaction()` for it doesn't really help with letting us change things about the implementation of Realm without breaking this, and having `transaction()` available will inevitably result in it being used in places that should be going through the Realm instance instead.

This code shouldn't actually be using a `Realm` instance at all, but with the GN going away entirely in v10 I don't think it's worth trying to redesign.

Reply:
Good points. :shipit: